compiler/resolution/ResolutionCandidate.cpp

```diff
@@ -273,6 +273,18 @@ bool ResolutionCandidate::computeAlignment(CallInfo& info) {
             skippedThisActual = true;
             skipNextActual = false;
             break;
+          } else if (formal->typeInfo() == dtMethodToken &&
+                     info.actuals.v[i]->typeInfo() == dtMethodToken) {
+            // both the actual and the formal were method tokens, but the this
+            // args could still be different.  Since this is an operator call,
+            // we don't actually care if the this args are different, so skip
+            // them, as well as the method tokens.
+            skipNextFormal = true;
+            formal = next_formal(formal);
+            j++;
+            skipNextActual = true;
+            skippedThisActual = true;
```

Comment:
Seems like you could write this section inside of if operator more simply as something like

```
if actual is method token
  skip next actual
if formal is method token
  skip next formal
if skip next actual
  ...
if skip next formal
  ...
```